app/uk/gov/hmrc/ngrpropertylinkingfrontend/controllers/CurrentRatepayerController.scala

```diff
@@ -59,14 +67,36 @@ class CurrentRatepayerController @Inject()(currentRatepayerView: CurrentRatepaye
       form
         .bindFromRequest()
         .fold(
-          formWithErrors => propertyLinkingRepo.findByCredId(CredId(request.credId.getOrElse(""))).flatMap{
-              case Some(property) =>  Future.successful(BadRequest(currentRatepayerView(createDefaultNavBar, formWithErrors, buildRadios(formWithErrors, ngrRadio), address = property.vmvProperty.addressFull, mode = mode)))
+          formWithErrors =>
+            //When validating from after apply, formError key is always empty. Bellow allows us to highlight the error field.
+            val correctedFormErrors = formWithErrors.errors.map(formError =>
+              if (formError.key.equals("")) {
+                formError.messages match
+                  case messages if messages.contains("currentRatepayer.day.empty.error") => formError.copy(key = "day")
+                  case messages if messages.contains("currentRatepayer.month.empty.error") => formError.copy(key = "month")
+                  case messages if messages.contains("currentRatepayer.year.empty.error") => formError.copy(key = "year")
+                  case _ => formError
+              } else {
+                formError
+              }
+            )
+            val formWithCorrectedErrors = formWithErrors.copy(errors = correctedFormErrors)
+            propertyLinkingRepo.findByCredId(CredId(request.credId.getOrElse(""))).flatMap{
+              case Some(property) =>  Future.successful(BadRequest(currentRatepayerView(createDefaultNavBar, formWithCorrectedErrors,
+                buildRadios(formWithCorrectedErrors, ngrRadio(formWithCorrectedErrors)), address = property.vmvProperty.addressFull, mode = mode)))
               case None => throw new NotFoundException("failed to find property from mongo")
```

Comment:
Slightly more preferable to throw this exception in the Repo class at source, so maybe see if we have a property first, if we don't then mongo throw the exception when it tries to get it, if it finds a property then we can go on to generating the view.